### ISSUE TITLE
[WeeklyReports] 2024.05.11~2024.05.24 周报汇总

### ISSUE THREAD

**Issue by AndSonder** [PFCCLab/Camp#280]
# [WeeklyReports] 2024.05.11~2024.05.24 周报汇总

请各位学员在本 issue 下以 comment 的形式填写周报摘要，ddl 本周五晚，格式示例如下：

```
### 姓名

xxx

### 本周工作

1. xxx
2. xxx
 
### 下周工作

1. xxx
2. xxx

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/xxx

```

#### 项目二：PIR 控制流专项

- [x] @huangjiyi (@winter-wang)

#### 项目三：PIR Python API 升级及机制建设

- [ ] @zrr1999 (@YuanRisheng, @0x45f)

#### 项目四：动转静 SOT 模块 Python 3.12 支持

- [x] @gouzil (@SigureMo)

#### 项目五：算子支持复数计算专项

- [ ] @zyt1024 (@GGBond8488)
- [x] @zbt78 (@GGBond8488)

#### 项目六：模型迁移工具建设

- [ ] @RedContritio (@zhwesky2010)

#### 项目七：框架 API 易用性提升

- [x] @NKNaN (@zhwesky2010)

#### 项目八：组合机制算子专项和机制建设

- [x] @zeroRains (@cyber-pioneer)
- [x] @Eddie-Wang1120 (@cyber-pioneer)

#### 项目九：高阶微分的性能分析和优化

- [x] @YibinLiu666 (@HydrogenSulfate)

#### 项目十：静态图半自动并行训练性能优化

- [x] @AndSonder (@From00)

#### 项目十一：全自动并行架构升级

- [x] @Corle-hyz (@Caozhou1995)

#### 项目十二：科学计算领域拓展专项（DeePMD-kit、光学案例）

- [x] @xusuyong (@HydrogenSulfate)

#### 项目十三：科学计算领域拓展专项（超分重构方向）

- [ ] @zhaojiameng (@lijialin03)

#### 项目十四：科学计算领域拓展专项（领域流体方向）

- [x] @DUCH714 (@wangguan1995)

#### 项目十五：科学计算领域拓展专项（领域气象方向）

- [x] @Yang-Changhui (@zhiminzhang0830)

#### 项目十六：CINN 支持动态 Shape 专项（前端方向）

- [x] @lshpku (@zyfncg, @jiahy0825)
- [ ] @ZelinMa557 (@zyfncg, @jiahy0825)

#### 项目十八：CINN 静态 shape 下鲁棒性和性能优化

- [x] @yulangz (@feifei-111, @2742195759)
- [ ] @WintersMontagne10335 (@feifei-111, @2742195759)

#### 项目十九：CINN 支持动态 Shape 专项（后端规则升级）

- [ ] @Austin-00 (@BiynXu)

#### 项目二十：CINN 支持动态 Shape 专项 （后端模型扩量）

- [x] @unseenme (@zhhsplendid)

#### 项目二十一：Paddle CMake 治理和编译优化

- [x] @silverling (@risemeup1)

#### 项目二十二：PaddleMIX 套件能力建设（文图方向）

- [x] @Tsaiyue (@westfish)

#### 项目二十三：PaddleMIX 套件能力建设（图文方向）

- [x] @cocoshe (@LokeZhou)

#### 项目二十四：大模型训练稳定性和高效低价小模型快速收敛

- [x] @lishuai-97 (@GuoxiaWang)

#### 项目二十六：推理 Pass & 融合算子优化

- [x] @yinfan98 (@yuanlehome)

#### 项目二十七：PaddleSpeech 套件能力建设

- [x] @kk-2000 (@zxcdlucy)
- [x] @mattheliu (@zxcdlucy)

**Comment by huangjiyi** [comment]
### 姓名

黄济懿

### 本周工作

1. 实现 YieldInstruction 以解决 YieldOp 的输入变量需要跳过 GC 所引发的内存泄露问题
   - https://github.com/PaddlePaddle/Paddle/pull/64234
2. 增加 WhileOp 中的 loop_vars 对 -1 shape 的支持
   - https://github.com/PaddlePaddle/Paddle/pull/64272
3. 规范用于自动生成算子定义的 yaml 文件路径
   - https://github.com/PaddlePaddle/Paddle/pull/64446
4. 将新 IR 下动静定义一致的算子迁移至 ops.yaml 和 backward.yaml
   - https://github.com/PaddlePaddle/Paddle/pull/64536

### 下周工作

- 推进 PIR 算子 yaml 定义规范化

### 详细周报链接
- https://github.com/PFCCLab/Camp/pull/284

**Comment by AndSonder** [comment]
### 姓名

卢畅

### 实习项目

静态图半自动并行训练性能优化

### 本周工作

#### 1. 为 ZBV 适配分布式标记

由于 ZBV 的 V 型编排和 VPP 的切图方式不同，我们需要对 ZBV 的分布式切分标记进行适配。


#### 2. 在实际业务场景下验证性能

在 Llama2 上进行了实际业务场景下的性能测试，发现了一些问题。

1. 当 chunk 是偶数的时候，计算 loss 的时间过长，导致了 bubble rate 过高

2. zbv 的官方时间中在没有 f 任务的时候会错误的插入 w 任务，导致了 b 任务的 delay


#### 3. 将 zbv 编排和显存估计功能结合


相关 PR：

- https://github.com/PaddlePaddle/Paddle/pull/63800
- https://github.com/PaddlePaddle/PaddleNLP/pull/8480


#### 4. 为流水线 timeline 工具适配动转静

之前流水线 timeline 可视化工具只能适配静态图，现在我们将其适配动转静的情况。

相关 PR：

- https://github.com/PaddlePaddle/Paddle/pull/64568
- https://github.com/PaddlePaddle/PaddleNLP/pull/8488

#### 详细周报链接

- https://github.com/PFCCLab/Camp/pull/286

**Comment by Yang-Changhui** [comment]
### 姓名

Yang-Changhui

### 本周工作

1. 完成earthformer技术文档的编写，并合入pr
2. 完成neuraloperator动态模型转静态模型的代码编写，并使用静态模型预测和可视化展示

### 下周工作

1. 根据导师意见，修改neuraloperator模型的代码，以及相关文档的编写

### 相关PR：

- https://github.com/PaddlePaddle/PaddleScience/pull/891
- https://github.com/PaddlePaddle/PaddleScience/pull/867

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/287

**Comment by lshpku** [comment]
### 姓名

梁书豪

### 本周工作

1. 在CINN后端实现新的Tiling流程
2. 排查生成的CUDA代码中访问越界的问题
 
### 下周工作

1. 将`CodeGenCUDA_Dev`中下标化简的逻辑移动到`PostProcess`中，并排查其他例外情况

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/282

**Comment by kk-2000** [comment]
### 姓名

张千芊

### 本周工作

1. 对demo进行测试
2. 整理当前可以修复的问题，相关issue：
- https://github.com/PaddlePaddle/PaddleSpeech/issues/3771
 
### 下周工作

1. 对当前可以修复的问题和BUG进行修复

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/260

**Comment by YibinLiu666** [comment]
### 姓名
YibinLiu666

### 本周工作
1. 收尾cumprod升级pr。
2. 定位到双向cumprod实现prod_grad的bug，静态图机制下cpu cumprod会自动使用inplace，原因未知。
3. 初步支持bmm的复数，complex64还有点问题

### 下周工作

1. 收尾双向cumprod实现prod_grad pr
2. 支持bmm复数类型

**Comment by NKNaN** [comment]
### 姓名

李睿文

### 本周工作

1. 针对 `paddle.nn.layer.state_dict` 的功能增强
2. 修复 `paddle.median` 的 min 分支存在的 bug
 
### 下周工作

1. 计划修改 paddle.io.BatchSampler：升级 sampler 参数，支持任意可迭代类型
2. 尝试进行升级 paddle.add/sub/div/mul 等二元 API 以支持 python number 的任务

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/290

**Comment by lishuai-97** [comment]
### 姓名

李帅

Github ID: [lishuai-97](https://github.com/lishuai-97)

### 本周工作

本周主要工作内容为完善XXX梯度裁剪算法论文，目前已完成论文投稿工作。
 
### 下周工作

总结实习项目，准备项目答辩。

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/292

**Comment by cocoshe** [comment]
### 姓名

叶柯

### 实习项目

PaddleMIX 套件能力建设（图文方向）

### 本周工作

本周主要工作如下：

优化qwen的mixtoken策略，解决了预处理数据集时显存易溢出的问题

+ https://github.com/PaddlePaddle/PaddleMIX/pull/548

### 下周工作

1. 推进pr合入

### 详细周报链接：
+ https://github.com/PFCCLab/Camp/pull/291

**Comment by gouzil** [comment]
### 姓名

田川

### 本周工作

1. 学习 torch 的`register_hook`的使用以及实现方法
2. 学习 tensorflow 的`py_function`的使用以及实现方法
3. 学习原有的`PyFuncOp`的实现方法
4. 初步构建`PyFuncOp`基本框架

pr: [#64624](https://github.com/PaddlePaddle/Paddle/pull/64624)

### 下周工作

1. 推进 pr 合入

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/293

**Comment by yulangz** [comment]
### 姓名

Github ID: [yulangz](https://github.com/yulangz)

### 实习项目

CINN子图鲁棒性和性能优化

### 本周工作

1. **CINN 故障修复**

    1. 修复 CINN Gather 算子在 2D Tensor 下的问题。PR: [https://github.com/PaddlePaddle/Paddle/pull/64474](https://github.com/PaddlePaddle/Paddle/pull/64474)
    2. 添加两个算子的 InferSymbolShape。PR：[https://github.com/PaddlePaddle/Paddle/pull/64653](https://github.com/PaddlePaddle/Paddle/pull/64653)

### 下周工作

1. 尝试解决 masked_select 结果形状未知的问题。

### 详细周报链接

https://github.com/PFCCLab/Camp/pull/294

**Comment by DUCH714** [comment]
### 姓名

DUCH714

### 本周工作

使用transformer结构有效地处理各种不同数据类型。
 
### 下周工作

尝试处理更多数据集，提升泛化能力。

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/296

**Comment by Eddie-Wang1120** [comment]
### 姓名
王晋恒

### 实习项目
组合机制算子专项和机制建设

### 本周工作
* https://github.com/PaddlePaddle/Paddle/pull/64425
* https://github.com/PaddlePaddle/Paddle/pull/64498
* https://github.com/PaddlePaddle/Paddle/pull/64381
* https://github.com/PaddlePaddle/Paddle/pull/64430
* https://github.com/PaddlePaddle/Paddle/pull/64036
* https://github.com/PaddlePaddle/Paddle/pull/64529 （clip decomp部分）
* https://github.com/PaddlePaddle/Paddle/pull/64021

### 下周工作

1. 继续进行算子拆解工作
2. 继续学习相关知识

### 详细周报链接

* https://github.com/PFCCLab/Camp/pull/288

**Comment by Tsaiyue** [comment]
### 姓名

蔡越

### 实习项目

PaddleMIX 套件能力建设（文图方向）

### 本周工作

1. **优化Open-Sora MultiheadCrossAttention注意力计算；**

   - 相关PR：https://github.com/PaddlePaddle/PaddleMIX/pull/537

2. **完成Open-Sora在notrainer下的训练代码的转写，并进行对齐；**

3. **实现基于PaddleNLP.trainer的训练流程，但在反向计算过程中出现精度不匹配问题，需进一步排查。**

### 下周工作

1. 排查基于trainer训练流程存在的问题，并在多卡环境下测试，整理提交训练相关PR；

**Comment by yinfan98** [comment]
### 姓名

尹帆

### 本周工作

1. RoPE kernel 支持 theta 参数 https://github.com/PaddlePaddle/PaddleNLP/pull/8440
2. blha 支持GQA https://github.com/PaddlePaddle/Paddle/pull/64640
 
### 下周工作

1. xxx
2. xxx

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/268

**Comment by xusuyong** [comment]
### 姓名

徐苏勇

### 本周工作

1. **修改NTK weight的PR**

	* 精度读文献 [When and why PINNs fail to train: A neural tangent kernel perspective](https://doi.org/10.1016/j.jcp.2021.110768)和[An Expert's Guide to Training Physics-informed Neural Networks](https://arxiv.org/abs/2308.08468)，参考作者开源的jax框架代码，比较paddle和jax的区别，使用paddlescience复现。合并PR。

   相关PR：[Add allen cahn sota](https://github.com/PaddlePaddle/PaddleScience/pull/879)
 
### 下周工作

1. xxx
2. xxx

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/298

**Comment by mattheliu** [comment]
### 姓名
刘卓鑫

### 实习项目
#### PaddleSpeech 套件能力建设

### 本周工作

1. **跑PaddleSpeech/Demos**
   **已跑完25/25**   **验证完成进度13/25**

### 下周工作

1. 修复Issue #3652 #3544 #3530
2. 修复demos/speech_server和demos/audio_content_search中存在的bug

### 详细周报链接
https://github.com/PFCCLab/Camp/pull/299

**Comment by silverling** [comment]
### 姓名

凌少鹏

### 本周工作

- 和导师讨论，重新规划 PHI 模块的架构：
  - https://github.com/PaddlePaddle/Paddle/pull/64541
 
### 下周工作

- 准备护航计划答辩

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/300

**Comment by unseenme** [comment]
### 姓名

Github ID: [unseenme](https://github.com/unseenme)

### 本周工作

1. 离线测试与筛选

2. 离线测试流程代码改进

3. 更新测试环境的框架版本

### 下周工作

1. 继续离线测试

2. 继续改进测试流程代码以进一步提高测试效率

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/281

**Comment by Corle-hyz** [comment]
### 姓名

何咏哲

### 本周工作

1. 借助显存公式完成动态均衡的拆分算法
2. 完善异构体系下的全自动并行方案，将其参数化
 
### 下周工作

1. 准备答辩

### 详细周报链接：

- https://github.com/PFCCLab/Camp/pull/295

**Comment by zbt78** [comment]
### 姓名

周波涛

### 实习项目

算子支持复数计算专项

### 本周工作

1. 合入inv相关工作 https://github.com/PaddlePaddle/Paddle/pull/63229
2. 完善之前pr

### 下周工作

1. 准备最终答辩
